tests/transactional/bootloader.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Basic smoke test to make sure that system can boot after
+#           few operations related to boot files and configurations
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use transactional;
+
+
+sub action {
+    my ($target, $text) = @_;
+    record_info('TEST', $text);
+    trup_call($target);
+    check_reboot_changes;
+}
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    action('bootloader',    'Reinstall bootloader');
+    action('setup-selinux', 'Enable SELinux') if (script_run("selinuxenabled") != 0);
```

Comment:
Is there a specific reason to do it in this test?

Simply scheduling this module after `enable_selinux` should have the same effect, especially because `setup-selinux` also does the same as `bootloader`.

Reply:
IMHO it is about showing that re-installing bootloader itself without enabled selinux does not break the system on any architecture. On s390x images and unless the initrd gets rebuilt, system become unbootable after enabling selinux.

Reply:
That sounds like this is more of a regression test for a very specific bug.

Can this be generalized for a use case which runs into the bug? I think that approach results in a better kind of module.